app/models/auth_sources/auth_source_ldap.rb

```diff
@@ -130,6 +131,20 @@ def users_in_group(name)
     ldap_con.user_list(name)
   end
 
+  def test_connection
+    result = {}
+    begin
+      Timeout::timeout(60) do
+        self.ldap_con.test
+      end
+      result[:success] = true
+      result[:message] = _("Test connection to LDAP server was successful.")
+    rescue Timeout::Error, Net::LDAP::Error, StandardError => exception
+      raise Foreman::Exception.new exception.message
```

Comment:
Nearly - use Foreman::WrappedException instead, which preserves the backtrace and other details from the original.